activerecord/lib/active_record/collection_cache_key.rb

```diff
@@ -2,30 +2,15 @@ module ActiveRecord
   module CollectionCacheKey
 
     def collection_cache_key(collection = all, timestamp_column = :updated_at) # :nodoc:
-      query_signature = Digest::MD5.hexdigest(collection.to_sql)
-      key = "#{collection.model_name.cache_key}/query-#{query_signature}"
+      model_signature = collection.model_name.cache_key
 
       if collection.loaded?
-        size = collection.size
-        timestamp = collection.max_by(&timestamp_column).public_send(timestamp_column)
+        unique_signature = collection.pluck(primary_key, timestamp_column).flatten.join("-".freeze)
```

Comment:
@sgrif Like @hdabrows said, `pluck` doesn't fire a query when the relation is loaded. [This test](https://github.com/rails/rails/pull/21503/files#diff-3809040490c52cb58e3f0fdfe6e90be0R43) in my PR covers that case too.